httpbakery/form/form.go

```diff
@@ -0,0 +1,144 @@
+// Package form enables interactive login without using a web browser.
+package form
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/juju/httprequest"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/environschema.v1"
+
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+)
+
+/*
+PROTOCOL
+
+A form login works as follows:
+
+	   Client                            Login Service
+	      |                                    |
+	      | GET visitURL with                  |
+	      | "Accept: application/json"         |
+	      |----------------------------------->|
+	      |                                    |
+	      |   Login Methods (including "form") |
+	      |<-----------------------------------|
+	      |                                    |
+	      | GET "form" URL                     |
+	      |----------------------------------->|
+	      |                                    |
+	      |                  Schema definition |
+	      |<-----------------------------------|
+	      |                                    |
+	+-------------+                            |
+	|   Client    |                            |
+	| Interaction |                            |
+	+-------------+                            |
+	      |                                    |
+	      | POST data to "form" URL            |
+	      |----------------------------------->|
+	      |                                    |
+	      |                Form login response |
+	      |<-----------------------------------|
+	      |                                    |
+
+The schema is provided as a environschema.Fileds object. It is the
+client's responsibility to interpret the schema and present it to the
+user.
+*/
+
+// Objects implementing the Handler interface interpret schema and
```

Comment:
Rather than HandleForm, I wonder about GetForm, making it more clear what the role of the method is. Handlers are usually server side.

	// FormGetter is implemented by clients to get the values for a form schema.
	// For example, an implementation might prompt the user for the value of
	// each field in the schema. The value returned from GetForm must be
	// compatible with the provided schema.
	type FormGetter interface {
	     GetForm(schema environschema.Fields) (map[string]interface{}, error)
	}

?